week-3/Homework/D-array-filter/exercise.js

```diff
@@ -6,14 +6,24 @@
   - Do not edit any of the existing code
 */
 
-var pairsByIndexRaw = [[0, 3], [1, 2], [2, 1], null, [1], false, "whoops"];
+var pairsByIndexRaw = [[0, 3], [1, 2], [2, 1], null, [1, 0], false, "whoops"];
+function filteredPairs(pair) {
+  return (
+    pair !== null &&
+    typeof pair === "object" &&
+    pair.length == 2 &&
+    pair[0] >= 0 &&
+    pair[1] >= 0
+  );
+}
 
```

Comment:
Thank you. I did not know about this method, it is very useful. Thank you so much.

Reply:
You only need to check two of the conditions above - since `Array.isArray()` will be `false` if `pair` is `null`. There's no reason the values couldn't be negative for the indexes, so I wouldn't check for that but if this were a real project it's the sort of thing I would probably want to ask whoever wrote the requirement.

Reply:
Thanks for your feedback. I removed excess conditions, now code looks more optimal. Thank you.